src/js/letters/actions/letters.js

```diff
@@ -27,49 +27,72 @@ import {
-            return dispatch({ type: BACKEND_SERVICE_ERROR });
-          case '403':
-            // Backend authentication problem
-            return dispatch({ type: BACKEND_AUTHENTICATION_ERROR });
-          case '502':
-            // Some of the partner services are down, so we cannot verify the
-            // eligibility of some letters
-            return dispatch({ type: LETTER_ELIGIBILITY_ERROR });
-          default:
-            return Promise.reject(
-              new Error(`vets_letters_error_server_get: ${error.status || 'unknown'}`)
-            );
-        }
+export function getLetterList(dispatch) {
+  return apiRequest(
+    '/v0/letters',
+    null,
+    response => dispatch({
+      type: GET_LETTERS_SUCCESS,
+      data: response,
+    }),
+    (response) => {
+      window.dataLayer.push({ event: 'letter-list-failure' });
+      if (typeof response.errors === 'undefined' || response.errors.length === 0) {
+        return Promise.reject(new Error('vets_letters_error_server_get: undefined error'));
       }
-    ).catch((error) => {
-      if (error.message.match('vets_letters_error_server_get')) {
-        Raven.captureException(error);
-        return dispatch({ type: GET_LETTERS_FAILURE });
+      const error = response.errors[0];
+      switch (error.status) {
+        case '503': // Handled same as 504
+        case '504':
+          // Either EVSS or a partner service is down or EVSS times out
+          return dispatch({ type: BACKEND_SERVICE_ERROR });
+        case '403':
+          // Backend authentication problem
+          return dispatch({ type: BACKEND_AUTHENTICATION_ERROR });
+        case '502':
+          // Some of the partner services are down, so we cannot verify the
+          // eligibility of some letters
+          return dispatch({ type: LETTER_ELIGIBILITY_ERROR });
+        default:
+          return Promise.reject(
+            new Error(`vets_letters_error_server_get: ${error.status || 'unknown'}`)
+          );
       }
-      throw error;
-    });
+    }
+  ).catch((error) => {
+    if (error.message.match('vets_letters_error_server_get')) {
+      Raven.captureException(error);
+      return dispatch({ type: GET_LETTERS_FAILURE });
+    }
+    throw error;
+  });
+}
+
+
+export function getBenefitSummaryOptions(dispatch) {
+  return apiRequest(
+    '/v0/letters/beneficiary',
+    null,
+    response => dispatch({
+      type: GET_BENEFIT_SUMMARY_OPTIONS_SUCCESS,
+      data: response,
+    }),
+    () => dispatch({ type: GET_BENEFIT_SUMMARY_OPTIONS_FAILURE })
+  );
+}
+
+
+// Call getLetterList then getBenefitSummaryOptions
+export function getLetterListAndBSLOptions() {
+  return (dispatch) => {
+    return getLetterList(dispatch)
+      // Maybe shouldn't try to get BSO options if we get an error or we don't get a BSL...
+      .then(() => getBenefitSummaryOptions(dispatch))
+      .catch(rejectedPromise => rejectedPromise); // getLetterList should return a rejected promise if it fails
```

Comment:
I think getLetterList actually returns a `dispatch({ type: GET_LETTERS_FAILURE })` _or_ throws an error (the rejected promise should get caught by the `catch()` on L#62). It might make sense to move the `catch()` for the `getLetterList()`'s `apiRequest()`into `getLetterListAndBSLOptions()`, so that we can consolidate top-level error handling and logging to Sentry (which is missing from `getBenefitSummaryOptions()` entirely).

Reply:
Oh blast. You're right. It _either_ returns the result of `dispatch()` _or_ throws the error. Missed that `return` before.

Yeah, I like the idea of moving that error handling logic up a level. Lemme see what I can come up with.